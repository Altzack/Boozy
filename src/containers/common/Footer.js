import React from 'react';
import styled from 'styled-components/macro';
import { FaGithub, FaLinkedin, FaTelegramPlane } from 'react-icons/fa';
import { MobileOnly, useResponsive } from './responsiveComponents';

const FooterSeparator = styled.span`
  padding: 0 10px;
`;

const FooterContainer = styled.div`
  width: 100%;
  color: rgba(232, 230, 227, 0.85);
  background-color: rgb(34, 36, 38);
  margin-top: 48px;
`;

const FooterContentContainer = styled.div`
  max-width: 1200px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: auto;
  width: 100%;
  min-height: 64px;
  ${({ mobile }) => mobile && 'flex-direction: column; padding: 24px 0px'}
`;

export default function Footer() {
  const { isTabletOrMobile } = useResponsive();

  return (
    <FooterContainer>
      <FooterContentContainer mobile={isTabletOrMobile}>
        <div>
          Boozy
          <FooterSeparator>|</FooterSeparator>
          Created by <i>Zack 2020</i>
        </div>
        <MobileOnly>
          <br />
        </MobileOnly>
        <div>
          <a href="https://github.com/Altzack">
            <FaGithub size="25px" style={{ marginRight: 15 }} />
          </a>
          <a href="https://www.linkedin.com/in/zack-altschuler">
            <FaLinkedin size="25px" style={{ marginRight: 15 }} />
          </a>
          <a href="mailto:altschulerzack@yahoo.com">
            <FaTelegramPlane size="25px" style={{ marginRight: 15 }} />
          </a>
        </div>
      </FooterContentContainer>
    </FooterContainer>
  );
}
