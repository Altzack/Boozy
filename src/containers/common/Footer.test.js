import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Footer from './Footer';
import setupTests from '../../setupTests';

it('renders a Footer by default', () => {
  const wrapper = shallow(<Footer />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
