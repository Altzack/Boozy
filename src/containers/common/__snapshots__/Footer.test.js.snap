// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Footer by default 1`] = `
<Footer__FooterContainer>
  <Footer__FooterContentContainer
    mobile={false}
  >
    <div>
      Boozy
      <Footer__FooterSeparator>
        |
      </Footer__FooterSeparator>
      Created by 
      <i>
        Zack 2020
      </i>
    </div>
    <MobileOnly>
      <br />
    </MobileOnly>
    <div>
      <a
        href="https://github.com/Altzack"
      >
        <FaGithub
          size="25px"
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
      </a>
      <a
        href="https://www.linkedin.com/in/zack-altschuler"
      >
        <FaLinkedin
          size="25px"
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
      </a>
      <a
        href="mailto:altschulerzack@yahoo.com"
      >
        <FaTelegramPlane
          size="25px"
          style={
            Object {
              "marginRight": 15,
            }
          }
        />
      </a>
    </div>
  </Footer__FooterContentContainer>
</Footer__FooterContainer>
`;
